homework/list.js

```diff
@@ -1,79 +1,149 @@
 'use strict';
 
-(function() {
+/**
+ * @typedef {Object} Note
+ * @property {String} title
+ * @property {String} passages
+ */
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+(function(exports) {
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
-    window.addEventListener('click', function(event) {
-      onNoteOpen(event);
-    });
+  /**
+   * Create a instance of ListManager
+   * @constructor
+   * @this {ListManager}
+   */
+  var ListManager = function(){
+      this._wrapper = document.querySelector('#note-list-wrapper');
+      this._listNoteContent = [];
   }
 
-  function onNoteOpen(event) {
-    if (event.target.classList.contains('note-title')) {
-      var id = event.target.dataset.noteId;
-      var content = _listNoteContent[id];
-      window.dispatchEvent(new CustomEvent('note-open',
-        { detail: content }));
-    };
-  }
 
-  function preloadFirstNote() {
-    if (_listNoteContent.length !== 0) {
-      var content = _listNoteContent[0];
+  /**
+   * Fetch the notes from remote server, draw them in the notes wrapper,
+   * and bind click event to dispatch note-open event
+   *
+   * @this {ListManager}
+   */
+  ListManager.prototype.start = function(){
+      this.fetchList().then((function(data) {
+        this.updateList(data);
+        this.drawList();
+        this.preloadFirstNote();
+      }).bind(this)).catch((function () {
+        // Deal with fetch error
```

Comment:
My slides apparently couldn't include all the things, or it will be at least 200+ pages and nobody want to see that. 
That's why I asked you to ask me everything you feel confused or unclear. And since clearly in the classroom and these 3 or 4 weeks nobody did that, I suppose you all know the things I have and haven't taught, until you and other individuals asked in private like this, which is acceptable for me.